### ISSUE TITLE
FSO repeats just one item from looped `+Misc anim group:` if animations are stored in animated png format.

### ISSUE THREAD

**Issue by JadedDragoon** [scp-fs2open/fs2open.github.com#3399]
This is a issue that has resurfaced during my attempts to apng-ify mjn's mainhalls. Here, a real world example:

From table\aquitaine-hall.tbm:
```
+Num Misc Animations:	14
+Misc anim:		2_aqu-m1a
+Misc anim:		2_aqu-m1b
+Misc anim:		2_aqu-m1c
+Misc anim:		2_aqu-m2a
+Misc anim:		2_aqu-m2b
+Misc anim:		2_aqu-m2c
+Misc anim:		2_aqu-m3a
+Misc anim:		2_aqu-m3b
+Misc anim:		2_aqu-m3c
+Misc anim:		2_aqu-m4
+Misc anim:		2_aqu-m5
+Misc anim:		2_aqu-m6a
+Misc anim:		2_aqu-m7a
+Misc anim:		2_aqu-m8
+Misc anim group:	1
+Misc anim group:	1
+Misc anim group:	1
+Misc anim group:	2
+Misc anim group:	2
+Misc anim group:	2
+Misc anim group:	3
+Misc anim group:	3
+Misc anim group:	3
+Misc anim group:	-1
+Misc anim group:	-1
+Misc anim group:	-1
+Misc anim group:	-1
+Misc anim group:	-1
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 9000 30000
+Misc anim delay:	-1 9000 30000
+Misc anim delay:	-1 9000 30000
+Misc anim delay:	-1 5000 9000
+Misc anim delay:	-1 5000 9000
+Misc anim delay:	-1 5000 9000
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
+Misc anim delay:	-1 0 0
```

There are three groups. Group 1 `/2_aqu-m1[abc]/` is the animation group for the lift and the man at the top of the lift. Group 2 `/2_aqu-m2[abc]/` is the animation group for the windows above the ready room doors and campaign room screen. Group 3 `/2_aqu-m3[abc]/` is the animation group for the large window on the left below the barracks and multiplayer doors and the fighter bay beyond it.

When the mainhall loads, one animation appears to be randomly selected from each group and is played after the associated delay. It then immediately repeats. No other animation from the group is played. There is no apparent delay between the animation reaching it's end and being repeated. Which animation gets played seems to change each time the mainhall is loaded.

This can actually be seen here: https://youtu.be/iKdcxRuE_X8?t=18 Watch the windows in the top right. `2_aqu-m2c` plays twice in a row with no delay and starts to play a third time before i open the campaign screen.

This is a test version of the mod containing just the Aquitane Mainhall and all animations converted to apng (including the full -hall.tbm from above: https://annapuma.scrapironcity.net/downloads/mjnmh-apng-1.0.0a.rar

**Comment by MjnMixael** [comment]
Further information. Hold type animations (Ones that play to the last frame and stay there) do work correctly. So this only affects group loops. 

In addition, this also seems to cause a debug assert related to sounds. I suspect related to too many loops with sounds called or something along those lines. I'm not really sure. Same mod, but EFFs instead of APNGs does not trigger this assert.
ASSERTION: "(val >= 0.0) && (val <= 1.0)" at sound.cpp:1229

**Comment by MageKing17** [comment]
Unfortunately, #3400 doesn't actually fix this; with some help from @ngld, I think I've come to a good enough understanding of the problem to attempt a proper fix, but it's a bit too involved to slip it at the end of an RC phase, so this won't get fixed in time for 21.2.
